File: src/main/java/pl/yalgrin/playnite/simplesync/web/CompanyResource.java
package pl.yalgrin.playnite.simplesync.web;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import pl.yalgrin.playnite.simplesync.dto.CompanyDTO;
import pl.yalgrin.playnite.simplesync.helper.SingleExecutorHelper;
import pl.yalgrin.playnite.simplesync.service.CompanyService;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("/api/company")
@RequiredArgsConstructor
public class CompanyResource {
    private final CompanyService service;
    private final SingleExecutorHelper singleExecutorHelper;

    @GetMapping("/{id}")
    public Mono<CompanyDTO> getCompany(@PathVariable Long id) {
        return service.findById(id);
    }

    @PostMapping("/save")
    public Mono<CompanyDTO> saveCompany(@RequestBody CompanyDTO dto, @RequestParam String clientId) {
        return singleExecutorHelper.runOnExecutor(service.saveObject(dto, clientId));
    }

    @PostMapping("/delete")
    public Mono<Void> deleteCompany(@RequestBody CompanyDTO dto, @RequestParam String clientId) {
        return singleExecutorHelper.runOnExecutor(service.deleteObject(dto, clientId));
    }
}

File: src/main/kotlin/pl/yalgrin/playnite/simplesync/web/library/CompanyResource.kt
package pl.yalgrin.playnite.simplesync.web.library

import org.springframework.web.bind.annotation.*
import pl.yalgrin.playnite.simplesync.dto.objects.CompanyDTO
import pl.yalgrin.playnite.simplesync.helper.SingleExecutorHelper
import pl.yalgrin.playnite.simplesync.service.objects.CompanyService
import reactor.core.publisher.Mono

@RestController
@RequestMapping("/api/company")
class CompanyResource(
    private val service: CompanyService,
    private val singleExecutorHelper: SingleExecutorHelper
) {

    @GetMapping("/{id}")
    fun getCompany(@PathVariable id: Long): Mono<CompanyDTO> {
        return service.findById(id)
    }

    @PostMapping("/save")
    fun saveCompany(@RequestBody dto: CompanyDTO): Mono<CompanyDTO> {
        return singleExecutorHelper.runOnExecutor(service.saveObject(dto))
    }

    @PostMapping("/delete")
    fun deleteCompany(@RequestBody dto: CompanyDTO): Mono<Void> {
        return singleExecutorHelper.runOnExecutor(service.deleteObject(dto))
    }
}